### PR TITLE
Add endpoint to return count of transactions missing decoding

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -1961,6 +1961,38 @@ Decode transactions that haven't been decoded yet
...
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      { "result": {"ethereum": 2, "optimism": 1, "base": 1}, "message": "" }
+
+   :resjson object result: A mapping of the chain name to the number of transactions missing the decoding.
```
Comment (LefterisJP) [inline docs/api.rst:1990]: This is just the evm chain name, right?

```suggestion
   :resjson object result: A mapping of the EVM chain name to the number of transactions missing the decoding.
```

File: docs/websockets.rst [hunk 2]
```diff
@@ -306,3 +306,25 @@ When we pull new rules from the remote repository and there are conflicts in the
...
+Transaction decoding process
+============================
+
+When the endpoint to start the task for decoding undecoded transactions is queried we send ws messages to inform about the progress.
+
+::
+    {
+        "type":"evm_decoding_pending_transactions",
+        "data":{
+            "chain":"ethereum",
```
Comment (LefterisJP) [inline docs/websockets.rst:320]: Didn't we say to call this evm_chain for consistency? Check other arguments in the same file. It's `evm_chain`.

```suggestion
            "evm_chain":"ethereum",
```

File: docs/websockets.rst [hunk 3]
```diff
@@ -306,3 +306,25 @@ When we pull new rules from the remote repository and there are conflicts in the
...
+            "total":2,
+            "processed":0
+        }
+    }
+
+- ``chain``: Evm chain where the task is decoding transactions.
```
Comment (LefterisJP) [inline docs/websockets.rst:326]:
```suggestion
- ``evm_chain``: Evm chain where the task is decoding transactions.
```

File: docs/websockets.rst [hunk 4]
```diff
@@ -306,3 +306,25 @@ When we pull new rules from the remote repository and there are conflicts in the
...
+- ``total``: Total number of transactions that will be decoded.
+- ``processed``: The total number of transactions that have been already decoded.
```
Comment (LefterisJP) [inline docs/websockets.rst:328]:
```suggestion
- ``processed``: The total number of transactions that have already been decoded.
```

File: docs/websockets.rst [hunk 5]
```diff
@@ -306,3 +306,25 @@ When we pull new rules from the remote repository and there are conflicts in the
...
+
+
+Transaction decoding process
+============================
+
+When the endpoint to start the task for decoding undecoded transactions is queried we send ws messages to inform about the progress.
+
+::
+    {
+        "type":"evm_decoding_pending_transactions",
```
Comment (LefterisJP) [inline docs/websockets.rst:318]:
```suggestion
        "type":"evm_undecoded_transactions",
```

File: rotkehlchen/api/websockets/typedefs.py [hunk 6]
```diff
@@ -24,6 +24,7 @@ class WSMessageType(Enum):
     REFRESH_BALANCES = auto()
     DATABASE_UPLOAD_RESULT = auto()
     ACCOUNTING_RULE_CONFLICT = auto()
+    EVM_DECODING_PENDING_TRANSACTIONS = auto()
```
Comment (LefterisJP) [inline rotkehlchen/api/websockets/typedefs.py:27]:
```suggestion
    EVM_UNDECODED_TRANSACTIONS = auto()
```

File: rotkehlchen/chain/evm/decoding/decoder.py [hunk 7]
```diff
@@ -545,9 +563,19 @@ def decode_transaction_hashes(
                 ignore_cache=ignore_cache,
             )
             events.extend(new_events)
-            if new_refresh_balances is True:
+            if new_refresh_balances is True and refresh_balances is False:
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/decoder.py:566]: why this change?

File: rotkehlchen/chain/evm/decoding/decoder.py [hunk 8]
```diff
@@ -527,7 +535,17 @@ def decode_transaction_hashes(
                 )
                 tx_hashes = [EVMTxHash(x[0]) for x in cursor]
 
-        for tx_hash in tx_hashes:
+        for tx_index, tx_hash in enumerate(tx_hashes):
+            if send_ws_notifications and tx_index % 10 == 0:
+                self.msg_aggregator.add_message(
+                    message_type=WSMessageType.EVM_DECODING_PENDING_TRANSACTIONS,
+                    data={
+                        'chain': self.evm_inquirer.chain_id.to_name(),
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/decoder.py:543]:
```suggestion
                        'evm_chain': chain_name,
```

File: rotkehlchen/chain/evm/decoding/decoder.py [hunk 9]
```diff
@@ -527,7 +535,17 @@ def decode_transaction_hashes(
                 )
                 tx_hashes = [EVMTxHash(x[0]) for x in cursor]
 
-        for tx_hash in tx_hashes:
+        for tx_index, tx_hash in enumerate(tx_hashes):
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/decoder.py:538]: move chain_id_name out of the loop
```suggestion
        chain_name = self.evm_inquirer.chain_id.to_name()
        for tx_index, tx_hash in enumerate(tx_hashes):
```

So that it does not run the `to_name() all the time in the loop + 1 in the end.

... or perhaps even better. Change `evm_inquirer.chain_name` to be derived from chain_id and not blockchain and use that?

File: rotkehlchen/chain/evm/decoding/decoder.py [hunk 10]
```diff
@@ -545,9 +563,19 @@ def decode_transaction_hashes(
...
             events.extend(new_events)
-            if new_refresh_balances is True:
+            if new_refresh_balances is True and refresh_balances is False:
                 refresh_balances = True
 
+        if send_ws_notifications:
+            self.msg_aggregator.add_message(
+                message_type=WSMessageType.EVM_DECODING_PENDING_TRANSACTIONS,
+                data={
+                    'chain': self.evm_inquirer.chain_id.to_name(),
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/decoder.py:573]:
```suggestion
                    'evm_chain': chain_name,
```

File: rotkehlchen/chain/evm/decoding/decoder.py [hunk 11]
```diff
@@ -527,7 +535,17 @@ def decode_transaction_hashes(
...
                 tx_hashes = [EVMTxHash(x[0]) for x in cursor]
 
-        for tx_hash in tx_hashes:
+        for tx_index, tx_hash in enumerate(tx_hashes):
+            if send_ws_notifications and tx_index % 10 == 0:
+                self.msg_aggregator.add_message(
+                    message_type=WSMessageType.EVM_DECODING_PENDING_TRANSACTIONS,
+                    data={
+                        'chain': self.evm_inquirer.chain_id.to_name(),
+                        'total': len(tx_hashes),
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/decoder.py:544]: same with len(tx_hashes). Calculate once out of the loop and use it everywhere.

File: rotkehlchen/db/evmtx.py [hunk 12]
```diff
@@ -287,7 +287,7 @@ def get_transaction_hashes_not_decoded(
     def count_hashes_not_decoded(
             self,
             chain_id: Optional[ChainID],
-            addresses: Optional[list[ChecksumEvmAddress]],
```
Comment (LefterisJP) [inline rotkehlchen/db/evmtx.py:290]: why make it optional? Explicit `=None` is cleaner code. Generaly avoid default args if possible.

File: rotkehlchen/chain/evm/decoding/decoder.py (hunk 7 above)
Comment (yabirgb) [reply rotkehlchen/chain/evm/decoding/decoder.py:566]: We are re-assigning the variable always. Made it just to avoid it

Comment (LefterisJP) [reply rotkehlchen/chain/evm/decoding/decoder.py:566]: makes no difference. The extra conditional check is more expensive than an assignment. Undo this.

File: rotkehlchen/chain/evm/decoding/decoder.py [hunk 13]
```diff
@@ -487,24 +487,32 @@ def get_and_decode_undecoded_transactions(
...
                 limit=limit,
                 addresses=addresses,
             )
-            self.decode_transaction_hashes(ignore_cache=False, tx_hashes=hashes)
+            self.decode_transaction_hashes(
+                ignore_cache=False,
+                tx_hashes=hashes,
+                send_ws_notifications=send_ws_notifications,
+            )
+            log.debug(f'Finished task decoding transactions for {self.evm_inquirer.chain_name} with {limit=}')  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/decoder.py:509]: Log needs to start with same phrase. Easy to grep/search for start/finish. Otherwise you need to tweak.

```suggestion
            log.debug(f'Finished task to process undecoded transactions for {self.evm_inquirer.chain_name} with {limit=}')  # noqa: E501
```

**Review by LefterisJP — APPROVED** [review]
nitpick can be done later
